src/Control.cpp

```diff
@@ -703,12 +727,59 @@ void Control::switchTo(Object* theTarget) {
         break;
       case kObjectNode: {
         //log.trace(kModControl, "Switching to node...");
-        audioManager.clear();
         if (currentNode()->isSlide())
           cameraManager.unlock();
         
         //log.trace(kModControl, "Set parent room");
+        Node* curNode = currentNode();
+        std::set<Audio*> curNodeAudios;
+
+        if (curNode && curNode->hasSpots()) {
+          curNode->beginIteratingSpots();
+
+          do {
+            Spot *curSpot = curNode->currentSpot();
+            if (curSpot->hasAudio() && curSpot->audio()->doesAutoplay()) {
```

Comment:
Should the auto flag of the Spot be used instead of the auto tag of the audio be used here?